packages/legacy/app/container-imp.tsx

```diff
@@ -1,25 +1,33 @@
+import { BaseLogger } from '@credo-ts/core'
 import { Container, TokenMapping } from '@hyperledger/aries-bifold-core'
 import { DependencyContainer } from 'tsyringe'
-
 export class AppContainer implements Container {
-  private container: DependencyContainer
-  public constructor(bifoldContainer: Container) {
-    this.container = bifoldContainer.getContainer().createChildContainer()
+  private _container: DependencyContainer
+  private log?: BaseLogger
+
+  public constructor(bifoldContainer: Container, log?: BaseLogger) {
+    this._container = bifoldContainer.container.createChildContainer()
+    this.log = log
+  }
+
+  public get container(): DependencyContainer {
+    return this._container
   }
+
   public init(): Container {
     // eslint-disable-next-line no-console
-    console.log(`Initializing App container`)
+    this.log?.info(`Initializing App container`)
     // Here you can register any component to override components in core package
     // Example: Replacing button in core with custom button
     // this.container.registerInstance(TOKENS.COMP_BUTTON, Button)
     return this
   }
 
   public resolve<K extends keyof TokenMapping>(token: K): TokenMapping[K] {
-    return this.container.resolve(token) as TokenMapping[K]
+    return this._container.resolve(token) as TokenMapping[K]
   }
 
-  public getContainer(): DependencyContainer {
-    return this.container
-  }
+  // public resolveSome<K extends keyof TokenMapping>(tokens: K[]): TokenMapping[K][] {
+  //   return tokens.map((token) => this.resolve(token))
+  // }
```

Comment:
Same q as below

Reply:
It was what I hoped work but it does not. It makes the TS type-checker sad. Will remove and save locally in case I feel like trying again at a later date.